CHANGES.rst

```diff
@@ -1066,6 +1066,9 @@ Other Changes and Additions
 - Updated the bundled CFITSIO library to 3.480. See
   ``cextern/cfitsio/docs/changes.txt`` for additional information. [#10256]
 
+- ``extract_array`` raises a ``ValueError`` if the data type of the
```

Comment:
Note for future self: This was discussed offline. In the end, it is neither API change or bug fix, as it clarifies the error message but does not change any behavior.